realm/realm-annotations-processor/src/test/resources/io/realm/SimpleRealmProxy.java

```diff
@@ -263,7 +263,7 @@ public static String getTableName() {
     }
 
     public static some.test.Simple copyOrUpdate(Realm realm, some.test.Simple object, boolean update, Map<RealmModel,RealmObjectProxy> cache) {
-        if (object instanceof RealmObjectProxy && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm().threadId != realm.threadId) {
+        if (object instanceof RealmObjectProxy && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm() != null && ((RealmObjectProxy) object).realmGet$proxyState().getRealm$realm().creatorThread != realm.creatorThread) {
```

Comment:
If you do that I think you lose the entire point of this refactor since you would introduce the method call back that was just removed?